lib/kino/control.ex

```diff
@@ -0,0 +1,126 @@
+defmodule Kino.Control do
+  @moduledoc """
+  Various widgets for user interactions.
+
+  Each widget is a UI control element that the user interacts
+  with, consequenty producing an event stream.
+
+  Those widgets are often useful paired with `Kino.Frame` for
+  presenting content that changes upon user interactions.
+
+  ## Examples
+
+  First, create a control and make sure it is rendered,
+  either by placing it at the end of a code cell or by
+  explicitly rendering it with `Kino.render/1`.
+
+      button = Kino.Control.button("Hello")
+
+  Next, to receive events from the control, a process needs to
+  subscribe to it and specify pick a name to distinguish the
+  events.
+
+      Kino.Control.subscribe(button, :hello)
+
+  As the user interacts with the button, the subscribed process
+  receives corresponding events.
+
+      IEx.Helpers.flush()
+      #=> {:event, :hello, %{origin: #PID<10895.9854.0>}}
+      #=> {:event, :hello, %{origin: #PID<10895.9854.0>}}
+  """
+
+  defstruct [:attrs]
+
+  @type t :: %__MODULE__{attrs: Kino.Output.control_attrs()}
+
+  defp new(attrs) do
+    token = Kino.Bridge.generate_token()
+    id = {token, attrs} |> :erlang.phash2() |> Integer.to_string()
+
+    attrs =
+      Map.merge(attrs, %{
+        id: id,
```

Comment:
We should actually have an unique id for subscriptions, I will adjust once the garbage collection is in place as per the other comment.

And for inputs we will need both `id` and reevaluation-safe-id (maybe `key` or `hash` :see_no_evil:).

Reply:
Sounds good to me!